### DISCUSSION TITLE
Checking for repo lock before execution

### DISCUSSION THREAD

**Discussion by cyb3rz3us** [borgbackup/borg#7045] [Q&A] [answered]
Hello Borg community --- I wanted to solicit feedback re: checking for repo lock prior to executing `borg` commands?

I am considering just checking for the existence of the `lock.exclusive` directory but I wanted to see if anyone else had some other methods...

**Comment by ThomasWaldmann** [comment, marked as answer]
borg checks by itself if the repo is locked. and you should not do stuff based on implementation details (like the directory you mentioned).

for some use cases, there is also `borg with-lock ...`.

**Reply by cyb3rz3us** [reply]
> borg checks by itself if the repo is locked

Understood but I would like to have some logic in my script that would detect the lock and either reschedule, wait, send an alert, etc.  I know there is already a wait timer common option but that only serves to delay or fail - no other actions are possible.  I suppose I will work with the RC and go from there.

> should not do stuff based on implementation details

I presume this is born out of concern that things could break due to changes in the implementation but frankly, anytime anyone builds a script or something to interface with a piece of software\hardware, it's always based on the implementation; the key is in trying to figure out which part of the implementation is the least likely to change.

Thanks for the feedback.
